### PR TITLE
Proposal: Add vsock support to KubeVirt

### PR DESCRIPTION
**Pull request by rmohr**
Proposal to add vsock support to KubeVirt.

gdoc which was used for initial discussions: https://docs.google.com/document/d/1KfFsO6jKvie9EGMl9FOMtfqAYAMUFOTl2nIup14besA/edit

### CODE REVIEW

**Comment by rthallisey** [comment]
/lgtm
Reviewed in the gdoc.  Proposal lgtm.

File: design-proposals/vsock/vsock.md
Comment (davidvossel) [inline design-proposals/vsock/vsock.md:65]: is there any chance that we'll need more fields associated with vsock other that "true|false" for attachment?  if so, maybe we shouldn't use a boolean.

Comment (rmohr) [reply design-proposals/vsock/vsock.md:65]: libvirt has three fields, one for auto-cid selection, one for the virtio model and one for manual cid assignment. We could not come up with additional use-cases where more fields would have to be on the VMI. For e.g. secure computing I would expect that vsock as a whole has to be disabled, but that's about it ...

Comment (rmohr) [reply design-proposals/vsock/vsock.md:65]: Can you think about something?

Comment (vladikr) [reply design-proposals/vsock/vsock.md:65]: Could that be that we will need more than one vsock per guest at some point?

Comment (rmohr) [reply design-proposals/vsock/vsock.md:65]: The vsock device has ports like TCP/IP. We can have a lot of agents over the single device. That is probably what you read about.

**Comment by davidvossel** [comment]
/approve
/hold

The community has already been discussing this design (in google doc form) for a few days before this pr was created, and no strong objections have been raised. I put a `hold` on the PR simply to give more people time to comment on this before it's merged. If there are no new open discussions by next Tuesday (Oct 18th), then I think it's reasonable to remove the hold.

File: design-proposals/vsock/vsock.md
Comment (xpivarc) [inline design-proposals/vsock/vsock.md:85]: Why did you choose containerd? I think containers/common would also benefit and might even reach a broader spectrum of users. (Kubervirt itself doesn't use containerd for some time...)

File: design-proposals/vsock/vsock.md
Comment (xpivarc) [inline design-proposals/vsock/vsock.md:92]: Nit `qemu-quest-agent` use case is mentioned above.

File: design-proposals/vsock/vsock.md
Comment (rmohr) [reply design-proposals/vsock/vsock.md:85]: Yes, but we use containerd. This is an independent effort of this proposal to secure it for our purpose. I think it makes a lot of sense to also push this to other popular CRIs, but it is out of scope of this proposal :)

File: design-proposals/vsock/vsock.md
Comment (rmohr) [reply design-proposals/vsock/vsock.md:92]: Could you explain what you mean? Running `qemu-guest-agent` on vsock is an explicit non-goal, but I wanted to outline how a possible future integration in a more general guest-agent mechanism could look like.

Comment (rmohr) [reply design-proposals/vsock/vsock.md:92]: Hm, they are two different points though:
 * This proposal does not change code-paths in virt-handler to allow running the qemu-guest-agent via vsock
 * Another point would be introducing a common polling mechanism, where no extra handling for the qemu-guest-agent would be there, it would use a common extensible mechanism.
 
 That are just some rough ideas to highlight that while qemu-guest-agent is not tackled  in this proposal, it is not in the way of doing so later.

File: design-proposals/vsock/vsock.md
Comment (xpivarc) [reply design-proposals/vsock/vsock.md:85]: I see. I read the proposal as a community - meaning we == community, and in that sense, it would make more sense to aim for changes in crio(therefore containers/common). I can pick this up.

File: design-proposals/vsock/vsock.md
Comment (xpivarc) [reply design-proposals/vsock/vsock.md:92]: Now I got you. I still feel like the last part of the second point is redundant. Your comment explained it very well, maybe we can incorporate it into the points?

File: design-proposals/vsock/vsock.md
Comment (rmohr) [reply design-proposals/vsock/vsock.md:85]: Ah, I see the confusion now regarding to the usage of `we`. :+1:

**Review by xpivarc — APPROVED** [review]

**Pull request merged**
